onnxruntime/core/optimizer/conv_add_fusion.cc

```diff
@@ -9,136 +9,108 @@ using namespace ONNX_NAMESPACE;
 using namespace ::onnxruntime::common;
 namespace onnxruntime {
 
-Status ConvAddFusion::ApplyImpl(onnxruntime::Graph& graph, bool& modified, int graph_level) const {
-  std::vector<onnxruntime::NodeIndex> removed_nodes;
-  for (auto& node : graph.Nodes()) {
-    ORT_RETURN_IF_ERROR(Recurse(node, modified, graph_level));
-
-    if (!graph_utils::IsSupportedOptypeVersionAndDomain(node, "Conv", 1) ||
-        !graph_utils::IsSupportedProvider(node, GetCompatibleExecutionProviders()) ||
-        node.GetOutputEdgesCount() != 1) {
-      continue;
-    }
+Status ConvAddFusion::Apply(Graph& graph, Node& node, bool& modified, bool& /*removed*/) {
+  auto& conv_node = node;
+  const auto& add_node = *conv_node.OutputNodesBegin();
+  const auto& conv_inputs = conv_node.InputDefs();
```

Comment:
If the Conv node produces graph output, would it have an empty output nodes list (so dereferencing the iterator would be an error)?

Reply:
I think we are safe because in the `SatisfyCondition` we do `node.GetOutputEdgesCount() != 1`. If it is a graph output it does not have an output edge.